src/content-feature.js

```diff
@@ -424,15 +425,211 @@ export default class ContentFeature {
+            constructorErrorMessage: 'Illegal constructor',
+            interfaceDescriptorOptions: { writable: true, enumerable: false, configurable: true, value: ImplClass },
+            wrapToString: true
+        }
+
+        const fullOptions = { ...defaultOptions, ...options }
+
+        // In some cases we can get away without a full proxy, but in many cases below we need it.
+        // For example, we can't redefine `prototype` property on ES6 classes.
+        // Se we just always wrap the class to make the code more maintaibnable
+
+        /** @type {ProxyHandler<Function>} */
+        const proxyHandler = {}
+
+        // handle the case where the constructor is called without new
+        if (fullOptions.allowConstructorCall) {
+            // make the constructor function callable without new
+            proxyHandler.apply = function (target, thisArg, argumentsList) {
+                return Reflect.construct(target, argumentsList, target)
+            }
+        }
+
+        // make the constructor function throw when called without new
+        if (fullOptions.disallowConstructor) {
+            proxyHandler.construct = function () {
+                throw new TypeError(fullOptions.constructorErrorMessage)
+            }
+        }
+
+        if (fullOptions.wrapToString) {
+            // mask toString() on class methods. `ImplClass.prototype` is non-configurable: we can't override or proxy it, so we have to wrap each method individually
+            for (const [prop, descriptor] of objectEntries(getOwnPropertyDescriptors(ImplClass.prototype))) {
+                if (prop !== 'constructor' && descriptor.writable && typeof descriptor.value === 'function') {
+                    ImplClass.prototype[prop] = new Proxy(descriptor.value, toStringProxyMixin(descriptor.value, `function ${prop}() { [native code] }`))
+                }
+            }
+
+            // wrap toString on the constructor function itself
+            Object.assign(proxyHandler, toStringProxyMixin(ImplClass, `function ${interfaceName}() { [native code] }`))
+        }
+
+        // Note that instanceof should still work, since the `.prototype` object is proxied too:
+        // Interface() instanceof Interface === true
+        // ImplClass() instanceof Interface === true
+        const Interface = new Proxy(ImplClass, proxyHandler)
+
+        // Make sure that Interface().constructor === Interface (not ImplClass)
+        if (ImplClass.prototype?.constructor === ImplClass) {
+            /** @type {StrictDataDescriptor} */
+            // @ts-expect-error - As long as ImplClass is a normal class, it should have the prototype property
+            const descriptor = getOwnPropertyDescriptor(ImplClass.prototype, 'constructor')
+            if (descriptor.writable) {
+                ImplClass.prototype.constructor = Interface
+            }
+        }
+
+        // mark the class as a shimmed class
+        objectDefineProperty(ImplClass, ddgShimMark, {
```

Comment:
I believe this is still visible in `getOwnPropertySymbols`, please validate :)

Reply:
Yeah, for sure. I thought it was an acceptable compromise so we can split `shimProperty` from `shimInterface` while still verifying the correct call order. Calling `shimProperty` before `shimInterface` would cause very tricky bugs.

We can also merge these two methods in one, what do you think?

Reply:
Yes, but it doesn't necessarily mean breakage. I do like your suggestion below to do it in test mode only though, so I'll do that.